src/main/java/uk/gov/companieshouse/chipsrestinterfacesconsumer/service/impl/MessageProcessorServiceImpl.java

```diff
@@ -37,33 +38,42 @@
     @Autowired
     private MessageProducer messageProducer;
 
+    @Autowired
+    private ConsumerConfig consumerConfig;
+
     @Override
-    public void processMessage(ChipsRestInterfacesSend message) throws ServiceException {
+    public void processMessage(ChipsRestInterfacesSend message, boolean isErrorAttempt) throws ServiceException {
         Map<String, Object> logMap = new HashMap<>();
         logMap.put("Message", message.getData());
         try {
             chipsRestClient.sendToChips(message);
         } catch (HttpStatusCodeException hsce) {
             logMap.put("HTTP Status Code", hsce.getStatusCode().toString());
-            handleFailedMessage(message, hsce, logMap);
+            handleFailedMessage(message, hsce, logMap, isErrorAttempt);
         } catch (Exception e) {
-            handleFailedMessage(message, e, logMap);
+            handleFailedMessage(message, e, logMap, isErrorAttempt);
         }
     }
 
-    private void handleFailedMessage(ChipsRestInterfacesSend message, Exception e, Map<String, Object> logMap) throws ServiceException {
+    private void handleFailedMessage(ChipsRestInterfacesSend message, Exception e, Map<String, Object> logMap, boolean isErrorAttempt) throws ServiceException {
```

Comment:
just thinking, rather than pass the boolean through the methods, could we not just inject the boolean config value into teh class? ${RUN_APP_IN_ERROR_MODE}

Reply:
or maybe make Application.runAppInErrorMode public

Reply:
although injecting ${RUN_APP_IN_ERROR_MODE} would be easier to test